phpBB/styles/prosilver/template/posting_buttons.html

```diff
@@ -5,6 +5,8 @@
 	var text_name = <!-- IF $SIG_EDIT -->'signature'<!-- ELSE -->'message'<!-- ENDIF -->;
 	var load_draft = false;
 	var upload = false;
+	var mention_url = '{UA_MENTION_URL}';
```

Comment:
I would prefer that these were handled via data attributes on a hidden input or other form of html tag to avoid using inline js cause eventually this will all be factored out. Might as well prep for it

Reply:
@hanakin so even very huge scripts such as the one in [plupload.html](https://github.com/phpbb/phpbb/blob/master/phpBB/styles/prosilver/template/plupload.html) will eventually be converted to `data-` attributes or something similar?
Requiring additional code to parse them back to the original JS objects?

I haven't made any performance checks or tests but using a script block seems to me as a rather convenient solution. Probably there should be one inline block at the bottom instead of many other, requiring some additional template features like INCLUDEJS for accumulating script data.

IMHO I'll better leave it as is right now so that it can be changed the same way together with other variables when that change rolls out.

Reply:
The aim would be to get rid of inline js all together.

Reply:
The new theme will probably use some build system for both JS and CSS by default, which will allow us to use some nice features (minification, auto-prefixing etc). This is obviously not usable on inlined JS or CSS.

Probably there will be some backend changes to support some features both for the new theme itself and extensions as well. However, these changes are not yet fully specified so we will probably publish our recommendations for extensions when we release those changes.

Reply:
@CHItA can't wait for those changes to be done :)
Are there any plans for a preprocessor (LESS/SASS) instead of plain CSS?

Reply:
There is never a reason to use inline scripts...just port it to its own file and include it using includejs. Using data attr for any template bar stuff etc... the use of minificatiin in the new theme coupled with the adoption of html2 will mean we can gloving most everything anyway at little cost. 
The new theme will be a completely different beast and will not be compatible with any extensions now nor should as it will be released under 4.0.0 meaning no compatibility. 

I am also looking into vuejs as an additional aspect as working with twig alone is problematic for certain circumstances where vue components just make it effortless and we gain the added benifit if less js for most of the interactivity.

Reply:
@hanakin the only reason I use inline scripts is exactly for template variables that cannot be inserted to the plain JS.
Almost all other JS stuff goes with `INCLUDEJS`.

Looking forward for new changes though.

Reply:
yes but you can instead use data for that and move everything to your js file.